Topics/Tech_Stacks/Vue_Component_Setup_and_Tips

```diff
@@ -0,0 +1,73 @@
+what the file format is, and also give general advice and things to watch out for. This guide assumes the reader has basic knowledge of HTML/CSS, as well as Javascript.
+
+If you have no knowledge of Vue or how to set up a Vue project, it is imperative that you follow [Vue's Quick Start](https://vuejs.org/guide/quick-start.html) instructions for doing so.
+
+## What is a Vue Component? Why should I care? <a name="what"></a>
+The Vue.js framework is a Javascript component-based programming model for the development of User Interfaces (UI), and is commonly used for Web Applications. It builds on the syntax of HTML, allowing the user to "declaratively" describe what they want the code to do. This allows for a very reactive app, one that responds efficiently to state changes to update the DOM. What this means is, less work needs to be done by the programmer for the HTML to update. The API takes care of a lot of it!
+
+Components are how it's all done. Every .vue file is a Vue Component! In these files, you can condense the 3 key aspects of a webpage, the HTML, code-behind, and CSS styling. This centralization and user-friendliness has made Vue renowned for its simplicity.
+
+**Am I quoting too much from official documentation? Should I put my own spin on it or cut out general info about the Vue.js framework?**
+
+> To learn more, please visit the [Vue website](https://vuejs.org/guide/introduction.html).
+
+## Format of Vue Components<a name="implementation"></a>
+Here is the general format of a Vue Component.
+
+
+```c
+# Example code
+<script setup>
+import { ref } from 'vue'
+const count = ref(0)
+</script>
+
+<template>
+  <button @click="count++">Count is: {{ count }}</button>
+</template>
+
+<style scoped>
+button {
+  font-weight: bold;
+}
+</style>
+```
+The top is the Javascript code-behind, the middle is the HTML, and the bottom is the CSS style. Nice and centralized. Thankfully, the order doesn't matter. However, as the script is the most complicated, we'll talk about it last.
+
+**Should I explain what the example code above means? Like, what import {ref} does? Or maybe I should give a different example, one similar to what we worked on in our CSC301 project? I worry it might be too long**
+
+### HTML Template
+Ever seen the HTML of a webpage? That and Vue's HTML template are almost identical. Just remember, the root container must be labelled as ```<template>```, and it can only have one container inside that. Inside *that* sub-container though, anything is possible! On top of things like your standard ```<div>```, you can also use **other** Vue Components that you've imported in the script section! More on that in a bit.
+
+Here, I'm going to *only* show how Vue HTML components have if-else visibility features, and the ability to insert components through built-in for loops. Am I missing anything important to do with HTML in Vue?
+
```

Comment:
Not as far as I know -- we basically just use HTML to define what the component renders to and to add a little logic like props/callbacks etc. But I'm not a Vue expert so grain of salt.